Update CategoryNotFoundException and Expense Equality

maxxyh opened the pull request:
Update codebase to use CategoryNotFoundException as a runtime exception.
Handle category not found exceptions separately at a command level by throwing CommandException.
Add checks for CategoryNotFoundException in ExpenseBook and JsonSerializableExpenseBook.


Tighten notion of Expense equality to include BOTH date and amount. 

Update test and data files to accommodate stricter checks of having to add a category before being able to add an expense to that category.

Fixes #141, fixes #121, fixes #101

maxxyh committed:
CategoryNotFoundException and Expense Equality

Update codebase to use CategoryNotFoundException as a runtime exception.
Handle category not found exceptions separately at a command level by throwing CommandException.
Add checks for CategoryNotFoundException in ExpenseBook and JsonSerializableExpenseBook.


Tighten notion of Expense equality to include BOTH date and amount. 

Update test and data files to accommodate stricter checks of having to add a category before being able to add an expense to that category.

maxxyh committed:
Fix checkstyle errors

Jaylenlee commented on src/main/java/seedu/expense/logic/commands/AddCommand.java:
```diff
@@ -29,10 +29,11 @@
-    public static final String MESSAGE_SUCCESS = "New expense added: %1$s";
-    public static final String MESSAGE_DUPLICATE_EXPENSE = "This expense already exists in the expense book";
+    public static final String MESSAGE_SUCCESS = "New expense added: %1$s.";
+    public static final String MESSAGE_DUPLICATE_EXPENSE = "This expense already exists in the expense book."
```
Add whitespace after fullstop?

maxxyh committed:
Standardised user-facing messages to have fullstops and whitespace

All have full-stops at the end except in the instances
where Expense#toString() is called.

Jaylenlee committed:
Merge branch 'master' into branch-HandleInvalidCategoryException-and-ExpenseEquality

maxxyh committed:
Fixed whitespace bug in AliasCommandTest

Jaylenlee commented:
LGTM
